internals/overlord/stateengine.go

```diff
@@ -72,6 +80,37 @@ func (se *StateEngine) State() *state.State {
 	return se.state
 }
 
+type startupError struct {
+	errs []error
+}
+
+func (e *startupError) Error() string {
+	return fmt.Sprintf("state startup errors: %v", e.errs)
+}
```

Comment:
Couldn't we replace `startupError`, `ensureError`, et al. with snapd's `multiError`?

Reply:
@anpep (cc @benhoyt) this is something I would do across the entire code base and separately, perhaps. So everything that has a chance of producing multiple errors uses a consistent approach. Also, we should consider leaving these changes as is if possible to make any future merge with the State Engine upstream easier.